controllers/construct/mongodbstatefulset.go

```diff
@@ -89,6 +89,9 @@ type MongoDBStatefulSetOwner interface {
 
 	// GetMongodConfiguration returns the MongoDB configuration for each member.
 	GetMongodConfiguration() map[string]interface{}
+
+	// NeedsAutomationConfigVolume returns whether the statefuslet needs to have a volume for the automationconfig.
+	NeedsAutomationConfigVolume() bool
```

Comment:
Community always returns `true`, but on AppDB it depends whether or not we are using vault